programs/mpl-core/src/plugins/verified_creators.rs

```diff
@@ -0,0 +1,205 @@
+    }
+
+    let mut result = SignatureChangeIndices {
+        added: Vec::new(),
+        changed: Vec::new(),
+        removed: Vec::new(),
+    };
+
+    for (i, sig) in new_verified_creators.signatures.iter().enumerate() {
+        match existing_map.get(&sig.address) {
+            Some(existing_sig) => {
+                if existing_sig.verified != sig.verified {
+                    result.changed.push(i as u8);
+                }
+            }
+            None => {
+                result.added.push(i as u8);
+            }
+        }
+    }
+
+    if let Some(verified_creators) = verified_creators {
+        for (i, sig) in verified_creators.signatures.iter().enumerate() {
+            if !new_signatures.contains(&sig.address) {
+                result.removed.push(i as u8);
+            }
+        }
+    }
+
+    Ok(result)
+}
+
+fn validate_verified_creators_as_creator(
+    new_verified_creators: &VerifiedCreators,
+    verified_creators: &VerifiedCreators,
+    authority: &Pubkey,
+) -> Result<ValidationResult, ProgramError> {
+    // Track any changes in verification status
+    let changes = calculate_signature_changes(new_verified_creators, Some(verified_creators))?;
+
+    if !changes.added.is_empty() || !changes.removed.is_empty() {
+        // creators cannot add new allowable signatures or remove existing ones
+        solana_program::msg!("Verified creators: Rejected");
+        return Err(MplCoreError::MissingSigner.into());
+    }
+
+    for change in changes.changed.iter() {
+        let sig = &new_verified_creators.signatures[*change as usize];
+        if &sig.address != authority {
+            // creators may only change their own verified status
+            solana_program::msg!("Verified creators: Rejected");
+            return Err(MplCoreError::MissingSigner.into());
+        }
+    }
+
+    Ok(ValidationResult::Approved)
+}
+
+fn validate_verified_creators_as_plugin_authority(
+    new_verified_creators: &VerifiedCreators,
+    verified_creators: Option<&VerifiedCreators>,
+    authority: &Pubkey,
+) -> Result<ValidationResult, ProgramError> {
+    // Plugin auth is allowed to change anything except the verified status of a signature to true
+    let changes = calculate_signature_changes(new_verified_creators, verified_creators)?;
+
+    for change in changes.changed.iter() {
+        let sig = &new_verified_creators.signatures[*change as usize];
+        if sig.verified && &sig.address != authority {
+            solana_program::msg!("Verified creators: Rejected");
+            return Err(MplCoreError::MissingSigner.into());
+        }
+    }
+
+    for addition in changes.added.iter() {
+        let sig = &new_verified_creators.signatures[*addition as usize];
+        if sig.verified && &sig.address != authority {
+            solana_program::msg!("Verified creators: Rejected");
+            return Err(MplCoreError::MissingSigner.into());
+        }
+    }
+
+    Ok(ValidationResult::Approved)
```

Comment:
I don't know why this validation approves, but the one for `Autograph` passes.  Really I think this one should return `Pass` as well and both `Autograph` and `VerifiedCreators` should be set to `CanReject` rather than `CanApprove` (see previous comment).